control-plane-operator/controllers/hostedcontrolplane/oapi/deployment.go

```diff
@@ -82,7 +93,30 @@ func oasContainerMain() *corev1.Container {
 	}
 }
 
-func buildOASContainerMain(image string) func(c *corev1.Container) {
+func oasKonnectivityProxyContainer() *corev1.Container {
+	return &corev1.Container{
+		Name: "oas-konnectivity-proxy",
+	}
+}
+
+func buildOASKonnectivityProxyContainer(routerImage string) func(c *corev1.Container) {
+	return func(c *corev1.Container) {
+		cpath := func(volume, file string) string {
+			return path.Join(volumeMounts.Path(c.Name, volume), file)
+		}
+		c.Image = routerImage
+		c.Command = []string{
+			"/bin/bash",
+			"-c",
+		}
+		c.Args = []string{
+			fmt.Sprintf("cat /etc/konnectivity-proxy-tls/tls.crt /etc/konnectivity-proxy-tls/tls.key > /tmp/tls.pem; haproxy -f %s", cpath(oasVolumeConfig().Name, oasKonnectivityProxyConfigKey)),
```

Comment:
use volumeMounts to get the paths to the cert/key, constant for the output pem file.